zerver/views/registration.py

```diff
@@ -83,21 +83,19 @@ def accounts_register(request: HttpRequest) -> HttpResponse:
         realm = None
     else:
         realm = get_realm(get_subdomain(request))
-        if prereg_user.realm is None:
-            return render(request, 'confirmation/link_expired.html')
-        if prereg_user.realm != realm:
```

Comment:
Link expiry comes from a settings.py setting, not from the realm. Or maybe I'm misunderstanding the question?

Reply:
A future simplification we should make to the confirmation code is to have get_object_from_key check that the subdomain matches the realm (and that the realm is not None), so that this works the way one would expect :).

Reply:
Discussed in person, and I've updated the commit message.